testing/citest/tests/azure_smoke_test.py

```diff
@@ -0,0 +1,242 @@
+    contract = jc.Contract()
+    return st.OperationContract(
+        self.agent.make_delete_app_operation(
+            application=self.TEST_APP,
+            account_name=self.bindings['SPINNAKER_AZURE_ACCOUNT']),
+        contract=contract)
+
+  def create_a_security_group(self):
+    """Creates AzContract for createServerGroup.
+
+    To verify the operation, we just check that the spinnaker security group
+    for the given application was created.
+    This will create a Network Security Group in a Resource Group on your Azure Subscription
+    """
+    rules = [
+        {
+            "access": "Allow",
+            "destinationAddressPrefix": "*",
+            "destinationPortRange": "80-80",
+            "direction": "InBound",
+            "endPort": 80,
+            "name": self.TEST_SECURITY_GROUP,
+            "priority": 100,
+            "protocol": "tcp",
+            "sourceAddressPrefix": "*",
+            "sourcePortRange": "*",
+            "startPort": 80
+        }]
+    job = [{
+        "provider": "azure",
+        "application": self.TEST_APP,
+        "appName": self.TEST_APP,
+        "region": self.bindings['TEST_AZURE_LOCATION'],
+        "stack": self.TEST_STACK,
+        "description": "Test - create security group for {app}".format(
+            app=self.TEST_APP),
+        "detail": "",
+        "credentials": self.bindings['SPINNAKER_AZURE_ACCOUNT'],
+        "securityRules": rules,
+        "name": self.TEST_SECURITY_GROUP,
+        "securityGroupName": self.TEST_SECURITY_GROUP,
+        "cloudProvider": "azure",
+        "type": "upsertSecurityGroup",
+        "user": "[anonymous]"
+    }]
+    builder = az.AzContractBuilder(self.az_observer)
+    (builder.new_clause_builder(
+        'Security Group Created', retryable_for_secs=30)
+     .collect_resources(
+         az_resource='network', 
+         command='nsg', 
+         args=['show', '--name', self.TEST_SECURITY_GROUP, '--resource-group', self.TEST_SECURITY_GROUP_RG])
+     .contains_pred_list([
+         jp.DICT_MATCHES({
+             'name': jp.STR_SUBSTR(self.TEST_SECURITY_GROUP)
+     })]))
+
+    payload = self.agent.make_json_payload_from_kwargs(
+        job=job, description=' Test - create security group for {app}'.format(
+            app=self.TEST_APP),
+        application=self.TEST_APP)
+
+    return st.OperationContract(
+        self.new_post_operation(
+            title='create_security_group', data=payload,
+            path='applications/{app}/tasks'.format(app=self.TEST_APP)),
+        contract=builder.build())
+
+  def delete_a_security_group(self):
+    """Creates azContract for deleteServerGroup.
+
+    To verify the operation, we just check that the spinnaker security group
+    for the given application was deleted.
+    """
+
+    job = [{
+            "Provider": "azure",
+            "appName": self.TEST_APP,
+            "region": self.bindings['TEST_AZURE_LOCATION'],
```

Comment:
Why did you choose to call this a location and not a region?
It looks like region is Azure terminology and looks like it is your data model terminology too.